docs/azure/role-assignments.md

```diff
@@ -0,0 +1,261 @@
+---
+title: Manage Azure role assignments
+description: Learn how to override Azure role assignments on .NET Aspire resources.
+ms.date: 03/31/2025
+---
+
+# Manage Azure role assignments
+
+All .NET Aspire Azure hosting integrations define Azure resources. [These resources](integrations-overview.md#add-azure-resources) come with default role assignments. You can replace these default role assignments with built-in role [or custom role assignments](integrations-overview.md#infrastructure-as-code). In this article, you learn how to manage Azure role assignments on .NET Aspire resources.
+
+## Default built-in role assignments
+
+When you add an Azure resource to the [app model](xref:dotnet/aspire/app-host#terminology), it's assigned default roles. If a resource depends on another resource, it inherits the same role assignments as the referenced resource unless explicitly overridden.
+
+Consider a scenario where an API project resource references an [Azure Search](../azureai/azureai-search-document-integration.md) resource. The API project is given the default role assignments, as shown in the following example:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var search = builder.AddAzureSearch("search");
+
+var api = builder.AddProject<Projects.Api>("api")
+                 .WithReference(search);
+```
+
+In the example code, the `api` project resource depends on the Azure `search` resource, meaning it references the `search` resource. By default, the `search` resource is assigned the following built-in roles:
+
+- <xref:Azure.Provisioning.Search.SearchBuiltInRole.SearchIndexDataContributor?displayProperty=nameWithType>
+- <xref:Azure.Provisioning.Search.SearchBuiltInRole.SearchServiceContributor?displayProperty=nameWithType>
+
+These role assignments allow the API project to read and write data to the Azure Search resource, and manage it. However, this behavior might not always be desirable. For instance, you might want to restrict the API project to only read data from the Azure Search resource.
+
+## Override default role assignments
+
+<!-- TODO: Add xref links when available for the WithRoleAssignments API. -->
+
+To override the default role assignment, use the `WithRoleAssignments` API and assign built-in roles as shown in the following example:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
```

Comment:
Your app needs to use a compute environment that supports role assignments. In 9.2 the only one we support is Azure Container Apps. So you need to call either:

* Aspire.Hosting.AzureContainerAppExtensions.AddAzureContainerAppEnvironment
* Aspire.Hosting.AzureContainerAppExtensions.AddAzureContainerAppsInfrastructure

In order for role assignments to work. If you don't enable ACA infrastructure, an exception is thrown when your app starts if you call `.WithRoleAssignments`.